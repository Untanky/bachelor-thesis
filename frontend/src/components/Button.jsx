import React from 'react';
import classnames from 'classnames';

import './Button.scss';

const Button = ({ type, onClick, children }) => {
  return (
    <button
      className={classnames('button', { cancel: type === 'cancel' })}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
