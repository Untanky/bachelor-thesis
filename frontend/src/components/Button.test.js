import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

it('renders cancel button correctly', () => {
  const tree = renderer
    .create(
      <Button type="cancel" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders normal button correctly', () => {
  const tree = renderer
    .create(
      <Button />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders delete button correctly', () => {
  const tree = renderer
    .create(
      <Button type="delete" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
