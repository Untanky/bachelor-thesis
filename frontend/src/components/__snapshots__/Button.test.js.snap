// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cancel button correctly 1`] = `
<button
  className="button cancel"
/>
`;

exports[`renders delete button correctly 1`] = `
<button
  className="button"
/>
`;

exports[`renders normal button correctly 1`] = `
<button
  className="button"
/>
`;
